### ISSUE TITLE
UIColor.hexString not supporting alpha

### ISSUE THREAD

**Issue by SuperWomble** [SwifterSwift/SwifterSwift#190]
Issue: hexString does not honour the color's alpha component.

Although the color's alpha is retrieved as part of the getting of the components, it is ignored, and  the output string is formatted to #RRGGBB instead of #RRGGBBAA

Desired outcome: if alpha is 1, then the current form should be returned. Otherwise, the string should be formatted to #(8 chars)
Cheers.

**Comment by SD10** [comment]
@SuperWomble Thank you for reporting this. Do you want to submit a pull request with this change?

**Comment by SuperWomble** [comment]
TBH, and I am shamed about this, but I'm a total github nub. I barely know how to read the site, let alone contribute anything more than issues. ::deep, deep shame::

**Comment by SD10** [comment]
@SuperWomble There's nothing to be ashamed about! I'm _still_ a git noob 😐

**Comment by pvtmert** [comment]
okay so i saw this and decided to close, but need review on following snippet:

```swift
extension UIColor {
	@available(*, deprecated, message: "Variable format no longer supported, use 'hexString(withAlpha:)' instead")
	public var hexString: String {
		return hexString(withAlpha: false)
	}

	public func hexString(withAlpha: Bool = false) -> String {
		let alphaShift = withAlpha ? 8 : 0
		var red:	CGFloat = 0
		var green:	CGFloat = 0
		var blue:	CGFloat = 0
		var alpha:	CGFloat = 0

		getRed(&red, green: &green, blue: &blue, alpha: &alpha)
		let rgb: Int = (withAlpha ? (Int)(alpha*255) : 0)
			| (Int)(red*255)   << (16+alphaShift)
			| (Int)(green*255) << (8+alphaShift)
			| (Int)(blue*255)  << (0+alphaShift)
		return NSString(format: (withAlpha ? "#%08x" : "#%06x"), rgb).uppercased as String
	}
}

let color = UIColor.init(red: 0.5, green: 0.5, blue: 0.5, alpha: 0.5)
print(color.hexString, color.hexString(withAlpha: true))
```

**Comment by LucianoPAlmeida** [comment]
@pvtmert Just one thing, the swift String class also has a constructor String(format: ), so you don't need to use NSString then cast it to string :)

**Comment by pvtmert** [comment]
@LucianoPAlmeida just tried in playground with set as ios, unfortunately it wants `[CVarArg]` but i supply `Int` so convert fails (even if i supply `[rgb]` for instance) so leaving as is like before...

**Comment by Alex293** [comment]
@SuperWomble shouldn't the format with transparency be #AARRGGBB ?

**Comment by pvtmert** [comment]
afaik in css it is called RGBA
here is ref: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value#rgba

**Comment by omaralbeik** [comment]
@pvtmert Would you please submit a pull request with the changes so we can discuss it there.

**Comment by Alex293** [comment]
@pvtmert sure in CSS but Android use ARGB (https://developer.android.com/guide/topics/resources/more-resources.html#Color).

In an app I would more often share values with Android than with web.

**Comment by pvtmert** [comment]
okay it has been 2 weeks since last comment, so lets decide on something what to do

ARGB = 👍
RGBA = 👎

okay lets start voting, i'll make pr after 604.800 seconds (week later)

**Comment by omaralbeik** [comment]
Fixed in #228, thanks to @pvtmert 🎉
